dandiapi/api/services/asset/__init__.py

```diff
@@ -107,6 +112,7 @@ def add_asset_to_version(
     asset_blob: AssetBlob | EmbargoedAssetBlob | None = None,
     zarr_archive: ZarrArchive | None = None,
     metadata: dict,
+    audit: bool = True,
```

Comment:
Would it help reduce the arguments to have a dataclass for data? Meaning could `asset_blob`, `zarr_archive` and `metadata` be grouped together?

Reply:
Is there a reason to exclude `user` and `version` from that dataclass?

Reply:
Also I understand this is breaking a ruff linting rule, but IMO, I don't think refactoring the arguments into a dataclass really solves anything. We'd still be passing the same arguments to the function, it would just be wrapped in a dataclass (and IMO would probably increase complexity while decreasing readability). So unless there is a convincing reason external to this linting error, I think we should just ignore that rule here.

Reply:
Fair enough. I was thinking that it could be used for most if not all of the functions in this file but I get it if it adds unnecessary complexity.

>Is there a reason to exclude user and version from that dataclass?
 
nope, just missed them